File: woodroid/src/main/java/me/gilo/woodroid/repo/APIMethod.java
package me.gilo.woodroid.repo;

import me.gilo.woodroid.models.Coupon;
import retrofit2.Call;

import java.util.List;

public interface APIMethod<T> {
    Call<T> create(T data);
    Call<T> get(int id);
    Call<List<T>> all();
    Call<T> update(int id, T data);
    Call<T> delete(int id);
    Call<T> delete(int id, boolean force);

}

File: woodroid/src/main/java/me/gilo/woodroid/repo/APIMethod.kt
package me.gilo.woodroid.repo


import retrofit2.Call

interface APIMethod<T> {
    fun create(data: T): Call<T>
    operator fun get(id: Int): Call<T>
    fun all(): Call<List<T>>
    fun update(id: Int, data: T): Call<T>
    fun delete(id: Int): Call<T>
    fun delete(id: Int, force: Boolean): Call<T>

}
